source/test_env.cpp

```diff
@@ -58,7 +64,13 @@ void notify_performance_coefficient(const char* measurement_name, const double v
 
 void notify_completion(bool success)
 {
-    printf("{{%s}}" NL "{{%s}}" NL, success ? TEST_ENV_SUCCESS : TEST_ENV_FAILURE, TEST_ENV_END);
+    printf("{{%s}}" NL, success ? TEST_ENV_SUCCESS : TEST_ENV_FAILURE);
+#ifdef YOTTA_CFG_DEBUG_OPTIONS_COVERAGE
+    coverage_report = true;
+    gcov_exit();
+    coverage_report = false;
```

Comment:
FH3 is only overridden between L69 and L71. Nothing else should be active during this period. We could disable interrupts here if necessary.